datadog_checks_base/datadog_checks/base/utils/time.py

```diff
@@ -0,0 +1,73 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from __future__ import absolute_import
+
+from datetime import datetime
+from time import time as epoch_offset
+
+import pytz
+from six import PY2
+
+UTC = pytz.utc
+EPOCH = datetime(1970, 1, 1, tzinfo=UTC)
+
+
+if PY2:
+
+    def get_timestamp(dt=None):
+        """
+        Returns the number of seconds since the Unix epoch.
+        If `dt` is not specified or `None`, the current time in UTC is assumed.
+        """
+        if dt is None:
+            return epoch_offset()
```

Comment:
`epoch_offset` give different precision in py2 and py3:

```
§ python3 -c 'from time import time as epoch_offset; print(epoch_offset())'
1589789968.9014442
§ python2 -c 'from time import time as epoch_offset; print(epoch_offset())'
1589789973.27
```

Reply:
Yup, I think https://www.python.org/dev/peps/pep-0564/#cpython-enhancements-of-the-last-5-years

> The CPython private "pytime" C API handling time now uses a new _PyTime_t type: simple 64-bit signed integer (C int64_t). The _PyTime_t unit is an implementation detail and not part of the API. The unit is currently 1 nanosecond.

Reply:
This might help us get the same precision for both py2 and py3: 

```
def datetime_to_float(d):
    epoch = datetime.datetime.utcfromtimestamp(0)
    total_seconds =  (d - epoch).total_seconds()
    # total_seconds will be in decimals (millisecond precision)
    return total_seconds
```

source: https://stackoverflow.com/a/35337826